openrtb_ext/convert_up_test.go

```diff
@@ -0,0 +1,382 @@
+package openrtb_ext
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/mxmCherry/openrtb/v16/openrtb2"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestConvertUpTo26(t *testing.T) {
+	testCases := []struct {
+		description     string
+		givenRequest    openrtb2.BidRequest
+		expectedRequest openrtb2.BidRequest
+		expectedErr     string
+	}{
+		{
+			description: "Malformed",
+			givenRequest: openrtb2.BidRequest{
+				Ext: json.RawMessage(`malformed`),
+			},
+			expectedErr: "req.ext is invalid: invalid character 'm' looking for beginning of value",
+		},
+		{
+			description: "2.4 -> 2.6",
+			givenRequest: openrtb2.BidRequest{
+				ID:   "anyID",
+				Ext:  json.RawMessage(`{"schain":{"complete":1,"nodes":[],"ver":"2"}}`),
+				Regs: &openrtb2.Regs{Ext: json.RawMessage(`{"gdpr":1,"us_privacy":"3"}`)},
+				User: &openrtb2.User{Ext: json.RawMessage(`{"consent":"1","eids":[{"source":"42"}]}`)},
+			},
+			expectedRequest: openrtb2.BidRequest{
+				ID:     "anyID",
+				Source: &openrtb2.Source{SChain: &openrtb2.SupplyChain{Complete: 1, Nodes: []openrtb2.SupplyChainNode{}, Ver: "2"}},
+				Regs:   &openrtb2.Regs{GDPR: openrtb2.Int8Ptr(1), USPrivacy: "3"},
+				User:   &openrtb2.User{Consent: "1", EIDs: []openrtb2.EID{{Source: "42"}}},
+			},
+		},
+		{
+			description: "2.4 -> 2.6 + Other Ext Fields",
+			givenRequest: openrtb2.BidRequest{
+				ID:     "anyID",
+				Ext:    json.RawMessage(`{"schain":{"complete":1,"nodes":[],"ver":"2"},"other":"otherExt"}`),
+				Source: &openrtb2.Source{Ext: json.RawMessage(`{"other":"otherSource","schain":{"complete":1,"nodes":[],"ver":"2"}}`)},
```

Comment:
Maybe you want to remove `"schain":{"complete":1,"nodes":[],"ver":"2"}` from the `Source.Ext` object since that is a 2.5 schain and this test is focused on the up convert from 2.4 to 2.6?

Reply:
Removed.